bot/validations.ts

```diff
@@ -1,18 +1,32 @@
+import { MainContext, OrderQuery, ctxUpdateAssertMsg } from "./start";
+import { ICommunity } from "../models/community";
+import { FilterQuery } from "mongoose";
+import { UserDocument } from "../models/user";
+import { IOrder } from "../models/order";
+import { Telegraf } from "telegraf";
+
 const { parsePaymentRequest } = require('invoices');
 const { ObjectId } = require('mongoose').Types;
-const messages = require('./messages');
-const { Order, User, Community } = require('../models');
+import * as messages from './messages';
+import { Order, User, Community } from '../models';
 const { isIso4217, isDisputeSolver } = require('../util');
 const { existLightningAddress } = require('../lnurl/lnurl-pay');
 const logger = require('../logger');
 
 // We look in database if the telegram user exists,
 // if not, it creates a new user
-const validateUser = async (ctx, start) => {
+const validateUser = async (ctx: MainContext, start: boolean) => {
   try {
-    const tgUser = ctx.update.callback_query
-      ? ctx.update.callback_query.from
-      : ctx.update.message.from;
+    let tgUser = null;
+    if (("callback_query" in ctx.update) && ctx.update.callback_query) {
+      tgUser = ctx.update.callback_query.from;
+    }
+    else if (("message" in ctx.update) && ctx.update.message) {
+      tgUser = ctx.update.message.from;
+    }
+    else {
+      throw new Error(ctxUpdateAssertMsg);
```

Comment:
@Mersho why are you changing behaviour here?

Reply:
@knocte Here's the [same thing](https://github.com/nblockchain/bot/pull/1#discussion_r1324273457)

Reply:
same thing? your link points me to a comment which points me to Zara&Parham's commit, but in that commit there's no `throw` addition

Reply:
nevermind, the part that threw was in a file of a diff which was collapsed by default; however, in Zara&Parham's commit, the throw happens in an `if`, not in an `else`; let's not do it in an else, errors should be thrown ASAP, the higher in the function the better

Reply:
> nevermind, the part that threw was in a file of a diff which was collapsed by default; however, in Zara&Parham's commit, the throw happens in an `if`, not in an `else`; let's not do it in an else, errors should be thrown ASAP, the higher in the function the better

If I remove the else statement, the function flow gets broken, so I don't know how to do that.

Reply:
I didn't ask you to remove the else statement, I asked you to put the throw in the `if` part instead of the `else` statement; to do that, you need to change the condition slightly